# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/Document.cpp
# Commit: bf27ab557da6
# Full Hash: bf27ab557da6f356bcaed60a596f77ac336b22c6
# Author: Ziran Sun <zsun@igalia.com>
# Date: 2023-05-23 21:47:35
# Regressor Bug: 1832837
# File Overlap Count: 1
# Description:
#   Bug 1832837 - Remove the fullscreen element ready check on fullscreen element descendant. r=edgar
#   
#   https://fullscreen.spec.whatwg.org/#fullscreen-element-ready-check
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D177913
# ==============================================================================

diff -r e117237fcb62 -r bf27ab557da6 dom/base/Document.cpp
--- a/dom/base/Document.cpp	Tue May 23 08:49:21 2023 +0000
+++ b/dom/base/Document.cpp	Tue May 23 08:51:37 2023 +0000
@@ -15286,15 +15286,6 @@
     aRequest.Reject("FullscreenDeniedHTMLDialog");
     return false;
   }
-  // XXXsmaug Note, we don't follow the latest fullscreen spec here.
-  //         This whole check could be probably removed.
-  if (fullscreenElement && !nsContentUtils::ContentIsHostIncludingDescendantOf(
-                               elem, fullscreenElement)) {
-    // If this document is fullscreen, only grant fullscreen requests from
-    // a descendant of the current fullscreen element.
-    aRequest.Reject("FullscreenDeniedNotDescendant");
-    return false;
-  }
   if (!nsContentUtils::IsChromeDoc(this) && !IsInFocusedTab(this)) {
     aRequest.Reject("FullscreenDeniedNotFocusedTab");
     return false;